doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1501,35 +1504,41 @@ Managing the local context
       like :g:`and` and :g:`exists` and those defined with the :cmd:`Record`
       command.
 
+.. todo PR: the following title seems inappropriate.  How about something
+   more like "Introducing new hypotheses", as in adding arbitrary terms rather
+   than transformations of existing terms??  But then I think the tactics in the
+   previous section (set, remember, pose, maybe decompose) should be moved into
+   this section.  But maybe hard to make the section seem like an crisp, intuitive grouping.
+   I can do the moving that after we've reviewed all the text.  WDYT?
```

Comment:
Yes, having `set`, `remember` and `pose` closer to the `pose proof`, `specialize`, et al. tactics would make sense, but I can also explain the current logic.

The tactics under the "Controlling the proof flow" logic are the ones which act on logical objects (propositions, which may be hypotheses or subgoals). One thing to note is that the tactics like `assert`, `enough`, `pose proof` or `specialize` are never required to complete a proof. They just allow to have a better control over how the proof is structured, to produce more readable proofs and to follow more closely the mathematical reasoning.

I don't have an alternative title that comes to mind right now, nor would I in case we moved `set`, `pose` and `remember` here. Maybe the solution is to also rethink the title of the previous section (if it were to stop after `rename`).

I propose to defer this question to after we've completed the review process for this PR (maybe something else worth talking about in a call).